packages/utils/src/fs/file-extension.ts

```diff
@@ -14,7 +14,7 @@ export const fileExtension = (resource: string): string => {
          */
         url = new URL(resource);
     } catch (err) {
-        return extname(resource).split('.')
+        return extname(resource ? resource.split('?')[0] : '').split('.')
```

Comment:
Should we have the additional spot fix here? We certainly don't need it to fix this specific bug since you've already updated that caller to pass an absolute URL. And if other callers have the same problem, wouldn't we prefer to update them to pass absolute URLs as well?

Reply:
> We certainly don't need it to fix this specific bug since you've already updated that caller to pass an absolute URL

I agree.

> And if other callers have the same problem, wouldn't we prefer to update them to pass absolute URLs as well

I prefer to have this fix for those scenarios. Passing a relative URL shouldn't be a problem, and that probably wasn't a problem at the beginning, because I think the old URL API supported relative urls.

Reply:
If you want to keep relative URLs supported, wouldn't it be less special-case to supply a dummy base URL on line 15? That would make relative URLs resolve so we can use the existing code path to get the file extension without having to maintain special handling in the `catch` clause.
```ts
url = new URL(resource, 'http://example.com');
```

Reply:
Sure, I can do that

Reply:
@antross done!